core/src/main/java/org/wso2/testgrid/core/command/CleanUpCommand.java

```diff
@@ -116,13 +126,21 @@ public void execute() throws CommandExecutionException {
             List<String> allTestPlans = testPlanUOW.deleteDatasourcesByAge(remainingBuildCount);
```

Comment:
This API is not correct. Why does an API like TestPlanUOW#deleteDatasourcesByAge return a list of test plans? Isn't it supposed to delete and return a true/false status?

Have you thought about memory requirements to load all old test plans?